src/storage/core.rs

```diff
@@ -303,8 +303,35 @@ where
             } else {
                 Err(e.into())
             }
-        })
+        });
+        self.try_update_active_blob(blob).await?;
+        result
+    }
+
+    async fn try_update_active_blob(&self, active_blob: &Box<Blob<K>>) -> Result<()> {
+        let config_max_size = self
+            .inner
+            .config
+            .max_blob_size()
+            .ok_or_else(|| Error::from(ErrorKind::Uninitialized))?;
+        let config_max_count = self
+            .inner
+            .config
+            .max_data_in_blob()
+            .ok_or_else(|| Error::from(ErrorKind::Uninitialized))?;
+        if active_blob.file_size() > config_max_size
+            || active_blob.records_count() as u64 > config_max_count
+        {
+            let now = std::time::SystemTime::now();
+            let dur = now.duration_since(active_blob.created_at())?;
+            if dur.as_millis() > self.inner.config.debounce_interval_ms() as u128 {
```

Comment:
SystemTime can easily be changed by user, so `dur.as_millis()` can easily be negative. I suggest to use `abs` when comparing